### ISSUE TITLE
Detect ESM/TypeScript rules defined with variable export

### ISSUE THREAD

**Issue by bmish** [not-an-aardvark/eslint-plugin-eslint-plugin#262]
We can detect rules defined with variables this in CJS but we are failing to do so with ESM/TypeScript:

```ts
import {Rule} from 'eslint';

const rule: Rule.RuleModule = {
  meta: {},

  create(context): Rule.RuleListener {
    const source = context.getSourceCode();
    // ...
  }
};

export = rule;
```

Example: https://github.com/43081j/eslint-plugin-lit/blob/master/src/rules/no-duplicate-template-bindings.ts

**Comment by aladdin-add** [comment]
but `export = rule` is not an esm(?). I'd suggest the users move to the standard esm(e.g. `export default rule`).

**Comment by bmish** [comment]
> ```ts
> import {Rule} from 'eslint';
> 
> const rule: Rule.RuleModule = {
>   meta: {},
> 
>   create(context): Rule.RuleListener {
>     const source = context.getSourceCode();
>     // ...
>   }
> };
> 
> export = rule;
> ```

Good point, updated title to reflect that this is only a TypeScript (not ESM) syntax, I still think we could support it.

https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

**Comment by aladdin-add** [comment]
ok, let's do it! 👍
